src/screens/BindEmailScreen/index.js

```diff
@@ -16,33 +21,85 @@ import {
 
 import Input from '@/components/Input';
 import ThemeButton from '@/components/ThemeButton';
+import PopupModal from '@/components/PopupModal';
+import LoadingSpinner from '@/components/LoadingSpinner';
+
+import useMailTimeSdk from '@/hooks/useMailTimeSdk';
 
 const BindEmailScreen = ({route, navigation}) => {
-  const [emails, setEmails] = useState(['', '']);
-  const [currentIndex, setCurrentIndex] = useState(0);
-  // const [bindEmailsRequest, { loading, error }] = useMutation(BIND_EMAIL_ACCOUNTS);
-
-  const handleUnbindEmailPress = index => {
-    emails.filter(email => email !== emails[index]);
-    Alert.alert(`success unbind email: ${emails[index]}`);
-    setEmails(emails.filter(email => email !== emails[index]));
-    setCurrentIndex(currentIndex - 1);
+  const [unbindSuccess, setUnbindSuccess] = useState(false);
+  const [clientError, setClientError] = useState('');
+  const {
+    login,
+    initiateSdk,
+    loading: sdkLoading,
+    error: sdkError,
+    loginSuccess,
+    loginFail,
+    loginCancel,
+  } = useMailTimeSdk();
+
+  const {authToken} = useContext(AuthContext);
+  const apiContext = {
+    context: {
+      headers: {
+        authorization: authToken ? `Bearer ${authToken}` : '',
+      },
+    },
+  };
+  const userEmailAccountsData = useQuery(
+    GET_USER_EMAIL_ACCOUNTS_API,
+    apiContext,
+  );
+  const [unbindEmailRequest, {loading}] = useMutation(
+    UNBIND_EMAIL_ACCOUNTS_API,
+    apiContext,
+  );
+
+  const currentIndex =
+    userEmailAccountsData?.data?.userProfile?.emailAccounts?.length || 0;
+
+  const emailAccounts = [
+    ...(userEmailAccountsData?.data?.userProfile.emailAccounts || []),
+    {id: null, emailAddress: ''},
+  ];
+
+  const [emails, setEmails] = useState(
+    emailAccounts || {id: null, emailAddress: ''},
+  );
+
+  useEffect(() => {
+    if (userEmailAccountsData?.data) {
+      setEmails(emailAccounts);
+    }
+  }, [emailAccounts, userEmailAccountsData]);
```

Comment:
I find that `emailAccounts` is not correct. Please change the code with following:
```suggestion
  const [emails, setEmails] = useState([{id: null, emailAddress: ''}]);

  useEffect(() => {
    if (userEmailAccountsData?.data) {
      const emailAccounts = [
        ...(userEmailAccountsData?.data?.userProfile.emailAccounts || []),
        {id: null, emailAddress: ''},
      ];
      setEmails(emailAccounts);
    }
  }, [userEmailAccountsData]);
```

Reply:
updated in cf08d97